core/api/service/chain/chain_api.hpp

```diff
@@ -0,0 +1,35 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef KAGOME_API_CHAIN_API_HPP
+#define KAGOME_API_CHAIN_API_HPP
+
+#include <boost/variant.hpp>
+#include "common/buffer.hpp"
+#include "outcome/outcome.hpp"
+#include "primitives/common.hpp"
+
+namespace kagome::api {
+
+  class ChainApi {
+   public:
+    virtual ~ChainApi() = default;
+    using BlockHash = kagome::primitives::BlockHash;
+    using ValueType = boost::variant<uint32_t, std::string>;
+
+    virtual outcome::result<BlockHash> getBlockHash() const = 0;
+
+    virtual outcome::result<BlockHash> getBlockHash(uint32_t value) const = 0;
+
+    virtual outcome::result<BlockHash> getBlockHash(
+        std::string_view value) const = 0;
```

Comment:
Is it hex string? value name is not representative

Reply:
fix in impl files as well

Reply:
refactored